doc/SONiC_static_route_hdl.md

```diff
@@ -0,0 +1,370 @@
+                +--rw identifier    -> ../config/identifier
+                +--rw name          -> ../config/name
+                +--rw static-routes
+                +--rw static* [prefix]
+                    +--rw prefix       -> ../config/prefix
+                    +--rw config
+                    |    +--rw prefix?  inet:ip-prefix
+                    +--ro state
+                    |  +--ro prefix?   inet:ip-prefix
+                    +--rw next-hops
+                      +--rw next-hop* [index]
+                         +--rw index                -> ../config/index ----(Read note below)
+                         +--rw config
+                         |  +--rw index?      string
+                         |  +--rw next-hop?   union
+                         |  +--rw metric?     uint32
+                         +--ro state
+                         |  +--ro index?      string
+                         |  +--ro next-hop?   union
+                         |  +--ro metric?     uint32
+                         +--rw interface-ref
+                         |  +--rw config
+                         |  |   +--rw interface?   -> /oc-if:interfaces/interface/name
+                         |  +--ro state
+                         |      +--ro interface?   -> /oc-if:interfaces/interface/name
+
+
+
+```
+
+
+Note: Each next-hop entry has a key called "index" in openconfig yang as shown above. This key is used to uniquely identify a entry in the list of next-hops. This key will be formed based on "interface (interface name if present) + next-hop (next-hop-ip if present) + vrf-name" and is expected to be provided correctly  by the user. Either interface name  or next-hop-ip or both should be present for any route to be accessed. i.e Ethernet0_2.2.2.2_default, 3.3.3.3_Vrf-RED. This key will not be stored in config DB.
+
+
+A new sonic-static-route.yang is defined to store entries in STATIC_ROUTE table in configDB.
+
+```diff
+
+module: sonic-static-route
+  +--rw sonic-static-route
+     +--rw STATIC_ROUTE
+        +--rw STATIC_ROUTE_LIST* [vrf-name prefix]
+           +--rw vrf-name   union
+           +--rw prefix     inet:ip-prefix
+           +--rw nexthop*   inet:ip-address
+           +--rw ifname*    union
+           +--rw distance*  uint32
+           +--rw nexthop-vrf* string
+
+
+```
+
+### 3.6.2 CLI
+#### 3.6.2.1 Configuration Commands
+#### 3.6.2.2 ip/ipv6 route config command
+ip route command is used to configure IPv4  static routes in SONiC.
+ipv6 route command is used to configure IPv6 static routes in SONiC.
+##### 3.6.2.2.1
+Syntax
+
+Vrf and distance metric are optional in the CLI.
+
+```
+ip route vrf <vrf-name> <prefix: A.B.C.D/mask> {[interface <interface-name>] | [<next-hop-ip>] | [<next-hop-ip> interface <interface-name>]} <distance Metric>
+```
+
+```
+ipv6 route vrf <vrf-name> <prefix: A.B.C.D/mask> {[interface <interface-name>] | [<next-hop-ip>] | [<next-hop-ip> interface <interface-name>]} <distance Metric>

```

Comment:
since distance is optional, can be put under []
Need to mention the default distance value

Reply:
sure, thanks.

Reply:
done